tests/ImageSharp.Web.Tests/Helpers/ExifOrientationUtilitiesTests.cs

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System.Numerics;
+using SixLabors.ImageSharp.Metadata.Profiles.Exif;
+using SixLabors.ImageSharp.Processing;
+using Xunit;
+
+namespace SixLabors.ImageSharp.Web.Tests.Helpers
+{
+    public class ExifOrientationUtilitiesTests
+    {
+        //                  150
+        // ┌─────┬─────┬─────┬─────┬─────┬─────┐
+        // │     │     │     │     │     │     │
+        // │     │    FFFFFFFFFFFFFFF    │     │
+        // ├─────X────F┌─────┬─────┬─────┼─────┤
+        // │     │    F│     │     │     │     │
+        // │     │    FFFFFFFFFFFFFFF    │     │
+        // ├─────┼────F┌─────┬─────┬─────┼─────┤ 100
+        // │     │    F│     │     │     │     │
+        // │     │    F│     │     │     │     │
+        // ├─────┼────F├─────┼─────┼─────┼─────┤
+        // │     │    F│     │     │     │     │
+        // │     │     │     │     │     │     │
+        // └─────┴─────┴─────┴─────┴─────┴─────┘
+        public static TheoryData<Vector2, Vector2, Vector2, ushort, Vector2> TransformVectorData =
+            new()
+            {
+                { new Vector2(25F, 25F), Vector2.Zero, new Vector2(150, 100), ExifOrientationMode.Unknown, new Vector2(25F, 25F) },
+                { new Vector2(25F, 25F), Vector2.Zero, new Vector2(150, 100), ExifOrientationMode.TopLeft, new Vector2(25F, 25F) },
+                { new Vector2(25F, 25F), Vector2.Zero, new Vector2(150, 100), ExifOrientationMode.TopRight, new Vector2(125F, 25F) },
+                { new Vector2(25F, 25F), Vector2.Zero, new Vector2(150, 100), ExifOrientationMode.BottomRight, new Vector2(125F, 75F) },
+                { new Vector2(25F, 25F), Vector2.Zero, new Vector2(150, 100), ExifOrientationMode.BottomLeft, new Vector2(25F, 75F) },
+                { new Vector2(25F, 25F), Vector2.Zero, new Vector2(150, 100), ExifOrientationMode.LeftTop, new Vector2(25F, 25F) },
+                { new Vector2(25F, 25F), Vector2.Zero, new Vector2(150, 100), ExifOrientationMode.RightTop, new Vector2(25F, 125F) },
+                { new Vector2(25F, 25F), Vector2.Zero, new Vector2(150, 100), ExifOrientationMode.RightBottom, new Vector2(75F, 125F) },
+                { new Vector2(25F, 25F), Vector2.Zero, new Vector2(150, 100), ExifOrientationMode.LeftBottom, new Vector2(75F, 25F) }
```

Comment:
That's not what this is testing nor should be. I'm testing two things.

 1. That the input values are scaled correctly from `25, 25` to `16.667, .25`. 
 2. That the input values are transformed to match a precalculated expected result. 

See the asci diagram [here](https://github.com/SixLabors/ImageSharp.Web/pull/241/files#diff-04782181c7482ecbfcbbed23d3eec6f4960985bc01828c070db3e24cb993a6acR13-R26). `25, 25` is marked `X` as the tested position

```
        //                  150
        // ┌─────┬─────┬─────┬─────┬─────┬─────┐
        // │     │     │     │     │     │     │
        // │     │    FFFFFFFFFFFFFFF    │     │
        // ├─────X────F┌─────┬─────┬─────┼─────┤
        // │     │    F│     │     │     │     │
        // │     │    FFFFFFFFFFFFFFF    │     │
        // ├─────┼────F┌─────┬─────┬─────┼─────┤ 100
        // │     │    F│     │     │     │     │
        // │     │    F│     │     │     │     │
        // ├─────┼────F├─────┼─────┼─────┼─────┤
        // │     │    F│     │     │     │     │
        // │     │     │     │     │     │     │
        // └─────┴─────┴─────┴─────┴─────┴─────┘
```

Should I be testing a position greater than the centre to ensure my previous issue is covered. Probably but `-25,-25` are non-sensical input values given the scaling bounds of `0,0` and `150,100`. 

TL;DR
The vector position MUST be within the Min/Max vector range.

Reply:
> The vector position MUST be within the Min/Max vector range.

I guess my current issue is that this isn't validated/enforced and the result is unclear. 

Wouldn't it make sense to throw an `ArgumentOutOfRangeException` if the position is outside of the min/max range:
```c#
if (position.X < min.X || position.Y < min.Y)
{
    throw new ArgumentOutOfRangeException(nameof(position), position, $"The position X and Y values should be greater than or equal to the min X and Y values ({min}).");
}

if (position.X > max.X || position.Y > max.Y)
{
    throw new ArgumentOutOfRangeException(nameof(position), position, $"The position X and Y values should be lower than or equal to the max X and Y values ({max}).");
}
```

Or if you don't want any exceptions in this code-path, enforce this by clamping the values by adding this to the top of the method and then use `normalized` instead:
```c#
var normalized = new Vector2(Math.Clamp(position.X, min.X, max.X), Math.Clamp(position.Y, min.Y, max.Y));
```

I'm fine with clamping the input value myself, but just wanted to point this out 😉

------------

From my testing, the following 'out of bounds' center positions currently result in the same 2 crops:
```
/Portrait_0.jpg?width=60&height=50&rxy=0,0
/Portrait_0.jpg?width=60&height=50&rxy=0,-1

/Portrait_0.jpg?width=60&height=50&rxy=0,1
/Portrait_0.jpg?width=60&height=50&rxy=0,2
```

But even through the resulting image is the same, because the URL is different they are cached as 2 separate files. This might not be a big deal, as that is currently also the case when using a negative width or height (and if both are 0 or lower, it will not even resize, but still cache the image), but still something to be aware of. And this can be manually worked-around by removing any invalid commands in the `OnParseCommandsAsync` callback, similar to how the maximum 4000x4000 size is handled...

Reply:
Thinking about this again, clamping the input value should probably be done in the (web)processor itself, just like the `rxy`/`CenterCoordinates` currently is.

The issue I pointed out is still valid though: any known commands are used to generate the cache key/hash, even if the command value is invalid, causing the same resulting image to be cached multiple times using different keys/hashes.

To make this a little bit more complex: different web processors can use the same commands (which might be the case for `width`, `height`, `orient` and `compand`), so command value sanitation isn't something a single processor should be responsible for. You don't want to remove a command that's invalid for one processor, but valid for another (and thus does result in a different result image).

Again, this isn't a big deal and can otherwise be worked around manually by removing any invalid command values using `OnParseCommandsAsync` (or in a more advanced setup, using an HMAC to validate whether the commands/values are generated by your own application).

Reply:
>The issue I pointed out is still valid though: any known commands are used to generate the cache key/hash, even if the command value is invalid, causing the same resulting image to be cached multiple times using different keys/hashes.

Yeah this is problematic. `OnParseCommandsAsync` is the best place to deal with it for the built in processors. A bounds check there would be sufficient. 

>To make this a little bit more complex: different web processors can use the same commands (which might be the case for `width`, `height`, `orient` and `compand`)

If someone is using width/height and have both `ResizeWebProcessor` and their own custom processor enabled they're in for a whole world of pain.

Reply:
I've added basic sanitation. 

https://github.com/SixLabors/ImageSharp.Web/pull/241/commits/444faa1b558463d134ced64603e56d80cae7abbb